File: backend/app/admin/service/internal/service/position.go
package service

import (
	"context"

	"github.com/go-kratos/kratos/v2/log"
	"google.golang.org/protobuf/types/known/emptypb"

	"kratos-monolithic-demo/app/admin/service/internal/data"

	"kratos-monolithic-demo/gen/api/go/common/pagination"
	v12 "kratos-monolithic-demo/gen/api/go/user/service/v1"
)

type PositionService struct {
	v12.UnimplementedPositionServiceServer

	log *log.Helper

	uc *data.PositionRepo
}

func NewPositionService(uc *data.PositionRepo, logger log.Logger) *PositionService {
	l := log.NewHelper(log.With(logger, "module", "position/service/admin-service"))
	return &PositionService{
		log: l,
		uc:  uc,
	}
}

func (s *PositionService) ListPosition(ctx context.Context, req *pagination.PagingRequest) (*v12.ListPositionResponse, error) {
	return s.uc.List(ctx, req)
}

func (s *PositionService) GetPosition(ctx context.Context, req *v12.GetPositionRequest) (*v12.Position, error) {
	return s.uc.Get(ctx, req)
}

func (s *PositionService) CreatePosition(ctx context.Context, req *v12.CreatePositionRequest) (*v12.Position, error) {
	return s.uc.Create(ctx, req)
}

func (s *PositionService) UpdatePosition(ctx context.Context, req *v12.UpdatePositionRequest) (*v12.Position, error) {
	return s.uc.Update(ctx, req)
}

func (s *PositionService) DeletePosition(ctx context.Context, req *v12.DeletePositionRequest) (*emptypb.Empty, error) {
	_, err := s.uc.Delete(ctx, req)
	if err != nil {
		return nil, err
	}
	return &emptypb.Empty{}, nil
}

File: frontend/apps/admin/src/rpc/service/position.ts
import type { PositionService } from '#/rpc/api/admin/service/v1/i_position.pb';
import type { Empty } from '#/rpc/api/google/protobuf/empty.pb';
import type { PagingRequest } from '#/rpc/api/pagination/v1/pagination.pb';
import type {
  CreatePositionRequest,
  DeletePositionRequest,
  GetPositionRequest,
  ListPositionResponse,
  Position,
  UpdatePositionRequest,
} from '#/rpc/api/user/service/v1/position.pb';

import { requestClient } from '#/rpc/request';

/** 职位管理服务 */
class PositionServiceImpl implements PositionService {
  async CreatePosition(request: CreatePositionRequest): Promise<Empty> {
    return await requestClient.post<Empty>('/positions', request);
  }

  async DeletePosition(request: DeletePositionRequest): Promise<Empty> {
    return await requestClient.delete<Empty>(`/positions/${request.id}`);
  }

  async GetPosition(request: GetPositionRequest): Promise<Position> {
    return await requestClient.get<Position>(`/positions/${request.id}`);
  }

  async ListPosition(request: PagingRequest): Promise<ListPositionResponse> {
    return await requestClient.get<ListPositionResponse>('/positions', {
      params: request,
    });
  }

  async UpdatePosition(request: UpdatePositionRequest): Promise<Empty> {
    const id = request.position?.id;
    if (request.position !== null) request.position.id = undefined;
    return await requestClient.put<Empty>(`/positions/${id}`, request);
  }
}

export const defPositionService = new PositionServiceImpl();
